### PR TITLE
[Merged by Bors] - feat: port Analysis.BoxIntegral.Basic

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/BoxIntegral/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,892 @@
...
+#align box_integral.integrable BoxIntegral.Integrable
+
+/-- The integral of a function `f` over a box `I` along a filter `l` w.r.t. a volume `vol`.  Returns
+zero on non-integrable functions. -/
+def integral (I : Box ι) (l : IntegrationParams) (f : ℝⁿ → E) (vol : ι →ᵇᵃ E →L[ℝ] F) :=
+  if h : Integrable I l f vol then h.choose else 0
+#align box_integral.integral BoxIntegral.integral
+
+-- Porting note: using the above notation ℝⁿ here causes the theorem below to be silently ignored
+-- see https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Lean.204.20doesn't.20add.20lemma.20to.20the.20environment/near/363764522
```
Comment (semorrison) [inline Mathlib/Analysis/BoxIntegral/Basic.lean:185]:
```suggestion
-- see https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Lean.204.20doesn't.20add.20lemma.20to.20the.20environment/near/363764522
-- and https://github.com/leanprover/lean4/issues/2257
```

File: Mathlib/Analysis/BoxIntegral/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,892 @@
...
+
+theorem to_subbox_aux (h : Integrable I l f vol) (hJ : J ≤ I) :
+    ∃ y : F, HasIntegral J l f vol y ∧
+      Tendsto (integralSum f vol) (l.toFilteriUnion I (Prepartition.single I J hJ)) (𝓝 y) := by
+  refine (cauchy_map_iff_exists_tendsto.1
+    (h.cauchy_map_integralSum_toFilteriUnion (.single I J hJ))).imp fun y hy ↦ ⟨?_, hy⟩
+  convert hy.comp (l.tendsto_embedBox_toFilteriUnion_top hJ)
+#align box_integral.integrable.to_subbox_aux BoxIntegral.Integrable.to_subbox_aux
+
+-- faster than `exact` here
```
Comment (semorrison) [inline Mathlib/Analysis/BoxIntegral/Basic.lean:547]: Is this comment stray? Perhaps it is referring to the `refine` in `to_subbox_aux`.

File: Mathlib/Analysis/BoxIntegral/Basic.lean [hunk 3]
```diff
@@ -0,0 +1,892 @@
...
+                  x ∈ Box.Icc J → (l.bDistortion → J.distortion ≤ c) → dist (vol J (f x)) (g J) ≤ ε)
+    (H₂ :
+      ∀ (c : ℝ≥0),
+        ∀ x ∈ Box.Icc I \ s,
+          ∀ ε > (0 : ℝ),
+            ∃ δ > 0,
+              ∀ J ≤ I,
+                Box.Icc J ⊆ Metric.closedBall x δ →
+                  (l.bHenstock → x ∈ Box.Icc J) →
+                    (l.bDistortion → J.distortion ≤ c) → dist (vol J (f x)) (g J) ≤ ε * B J) :
```
Comment (semorrison) [inline Mathlib/Analysis/BoxIntegral/Basic.lean:758]: Maybe we don't need quite all of the newlines here!

File: Mathlib/Analysis/BoxIntegral/Basic.lean [hunk 4]
```diff
@@ -0,0 +1,892 @@
...
+                  x ∈ Box.Icc J → (l.bDistortion → J.distortion ≤ c) → dist (vol J (f x)) (g J) ≤ ε)
+    (H₂ :
+      ∀ (c : ℝ≥0),
+        ∀ x ∈ Box.Icc I \ s,
+          ∀ ε > (0 : ℝ),
+            ∃ δ > 0,
+              ∀ J ≤ I,
+                Box.Icc J ⊆ Metric.closedBall x δ →
+                  x ∈ Box.Icc J →
+                    (l.bDistortion → J.distortion ≤ c) → dist (vol J (f x)) (g J) ≤ ε * B J) :
```
Comment (semorrison) [inline Mathlib/Analysis/BoxIntegral/Basic.lean:862]: Similarly.

**Comment by semorrison** [comment]
bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**